### ISSUE TITLE
Date picker is completely inaccessible

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#14141]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The date picker is impossible to use with a screen reader and/or keyboard only.  I can tell from the code it's supposed to be opening a picker dialog, but either that is never triggered or it cannot be navigated to by keyboard.

Looking at the [`app-datepicker` demos](https://motss.xyz/demos/app-datepicker), the picker can certainly be made more accessible, but those demos are still not very keyboard friendly.  For example, the input field version is read only so you cannot enter manually, the years are just presented as a long list of buttons so you cannot jump to one easily, etc.

Similar to `ha-combo-box`, I would suggest switching to [`vaadin-date-picker-light`](https://cdn.vaadin.com/vaadin-web-components/23.3.0-alpha2/#/elements/vaadin-date-picker-light) which is far more accessible right out of the box.  That solution would probably also close the other issues mentioned below.

### Describe the behavior you expected

I expect to be able to easily select a date or enter one manually with keyboard only, and without an exorbitant amount of navigating inside the picker.

### Steps to reproduce the issue

1. Create a date helper
2. Open in more info or put it on a dashboard card
3. Try to enter a date with only the keyboard

### What version of Home Assistant Core has the issue?

2022.10.x

### In which browser are you experiencing the issue with?

Any

### Which operating system are you using to run this browser?

Any

### Additional information

- #13582 
- #13278 
- #12329

**Comment by steverep** [comment]
Still very much an issue

**Comment by steverep** [comment]
Another good reason to remove `app-datepicker` is that it's slowing things down at least in CI.  It includes `nodemod`, which then includes `lit-ntml`, which then includes `@reallyland/esm`.  They all need building every time we run `yarn install`.

cc: @bramkragten

**Comment by steverep** [comment]
Still relevant

**Comment by alex3305** [comment]
I'd like to add to this that the time inputs currently don't show a highlight, which hinders accessibility and keyboard usage. Because there is no indication where my cursor is:
![image](https://github.com/home-assistant/frontend/assets/5155694/cf02948a-082c-44d8-b6e4-33564214962b)

In the screenshot above my cursor is in the top left corner. An alternative background color:
![image](https://github.com/home-assistant/frontend/assets/5155694/bb757c2a-aefd-44c1-9a88-beabdaf9528b)
(this is with `background: var(--mdc-text-field-fill-color);`) or a border should be sufficient to make this bit keyboard accessible.

I know it's not perfect, but it would be a little step 😉

**Comment by wendevlin** [comment]
@steverep would you have time to test the new date picker that is based on cally?

**Comment by jamesmyatt** [comment]
Current date picker also makes it very hard to pick a date that isn't within a few months of today. See https://github.com/home-assistant/core/issues/167392.

**Comment by wendevlin** [comment]
> Current date picker also makes it very hard to pick a date that isn't within a few months of today. See [home-assistant/core#167392](https://github.com/home-assistant/core/issues/167392).

Thanks James, but this issue is about accessibility for screen readers, for users with disabilities. You talk about the user experience of the date picker. Makes totally sense. We may fix it in the future 👍

**Comment by jamesmyatt** [comment]
Thanks @wendevlin . Yes, and I suspect that a more accessible date picker will benefit all users by being more usable.

**Comment by skragzombi666** [comment]
Indeed. Still very much an issue!
